src/foundry/client/apps/sidebar/apps/compendium.d.mts

```diff
@@ -9,7 +9,7 @@ declare global {
   class Compendium<
     Metadata extends CompendiumCollection.Metadata,
     Options extends ApplicationOptions = ApplicationOptions,
```

Comment:
Assuming for the new `Options` definition it should be `Options extends Compendium.Options<Metadata> = Compendium.Options<Metadata>`, (ie the first `Compendium.Options` type should be `Metadata` and not `Options`)?

And then for the interface definition we have it extend `Partial<DocumentDirectoryOptions>` because the constructor could contain an empty object? If so not sure how to modify the `extends DocumentDirectory<Options>` part, as I don't think it wants any part of `Options` to be optional

Reply:
Yes on the first part

For the second, right now the constructor takes `options: Partial<Options>`. What I'm doing is making the `compendium` property mandatory; the end result for `Options` is all of the inherited options from `DocumentDirectoryOptions` are still optional, but `compendium` is mandatory. One thing that's often helpful is just opening up the `tests` folder and either finding or making the appropriate `compendium.test-d.ts` file, and seeing how different implementations change the expected properties.

Reply:
I think I understand the second part. But I don't think they _can_ be optional, as the implementation of the `Compendium` class extends `DocumentDirectory`, and they're not optional for `DocumentDirectory`; when I try with that, it tells me "Property 'baseApplication' is optional in the type 'Options<Metadata>' but required in type 'DocumentDirectoryOptions'." (and if I add `baseApplication`, then it's "Property 'width'" etc.

Reply:
Ah, so be it. Just `extend DocumentDirectoryOptions`; you technically *can* have an empty compendium, it's just not very helpful. And then the constructor is still `options: Partial<Options>`